pkgs/tools/misc/archi/default.nix

```
@@ -58,13 +61,31 @@ stdenv.mkDerivation rec {
           --prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath ([ glib webkitgtk_4_0 ])} \
           --set WEBKIT_DISABLE_DMABUF_RENDERER 1 \
           --prefix PATH : ${jdk}/bin
+
+        install -Dm444 icon.xpm $out/share/icons/hicolor/256x256/apps/archi.xpm
+
+        runHook postInstall
       ''
     else
       ''
         mkdir -p "$out/Applications"
         mv Archi.app "$out/Applications/"
       '';
```

```suggestion
        runHook postInstall
      ''
    else
      ''
        runHook preInstall

        mkdir -p "$out/Applications"
        mv Archi.app "$out/Applications/"

        runHook postInstall
      '';
```
Please also add preInstall a few lines above